test/NSHM_NZ/inversion/CrustalFaultSectionTest.java

```diff
@@ -0,0 +1,282 @@
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import org.opensha.commons.data.CSVFile;
+import org.opensha.commons.geo.Location;
+import org.opensha.commons.util.IDPairing;
+import org.opensha.commons.util.XMLUtils;
+import org.opensha.refFaultParamDb.vo.FaultSectionPrefData;
+import org.opensha.sha.faultSurface.FaultSection;
+import org.opensha.sha.faultSurface.FaultTrace;
+
+import scratch.UCERF3.enumTreeBranches.DeformationModels;
+import scratch.UCERF3.enumTreeBranches.FaultModels;
+import scratch.UCERF3.enumTreeBranches.ScalingRelationships;
+import scratch.UCERF3.enumTreeBranches.SlipAlongRuptureModels;
+import scratch.UCERF3.inversion.coulomb.CoulombRates;
+import scratch.UCERF3.inversion.InversionFaultSystemRupSet;
+import scratch.UCERF3.inversion.laughTest.UCERF3PlausibilityConfig;
+import scratch.UCERF3.inversion.SectionCluster;
+import scratch.UCERF3.inversion.SectionClusterList;
+import scratch.UCERF3.logicTree.LogicTreeBranch;
+import scratch.UCERF3.utils.DeformationModelFetcher;
+import scratch.UCERF3.utils.FaultSystemIO;
+
+import scratch.UCERF3.inversion.UCERF3SectionConnectionStrategy;
+
+/*
+ * discarded ---
+*/
+
+
+
+public class CrustalFaultSectionTest {
+
+	private static final double grid_disc = 5d;
+
+	
+	static List<FaultSection> fsd;
+	
+	@BeforeClass
+	public static void setUpBeforeClass() throws Exception {
+		// this is the input fault section data file
+		File fsdFile = new File("./test/NSHM_NZ/inversion/fixtures/alderman_sections.xml");
+		// load in the fault section data ("parent sections")
+		fsd = FaultModels.loadStoredFaultSections(fsdFile);
+	}
+	
+	@Test
+	public void testXMLSectionsLoaded() throws IOException {
+		// rake value/s should have no effect on output without plausibility filter 
+		
+		assertEquals(16, fsd.size());
+		
+	}
+	
+    @Test 
+	public void testRuptureGeneratorSetup() throws IOException {
+
+		
+		System.out.println(fsd.size() + " Crustal Fault Sections");
+
+		// directory to write output files
+		File outputDir = new File("/tmp");
+		// maximum sub section length (in units of DDW)
+		double maxSubSectionLength = 0.5;
+		// max distance for linking multi fault ruptures, km
+		double maxDistance = 10d;
+
+		// write crustal section data to file
+		File subductionSectDataFile = new File(outputDir, "crustal_sections.xml");
+		Document doc0 = XMLUtils.createDocumentWithRoot();
+		FaultSystemIO.fsDataToXML(doc0.getRootElement(), FaultModels.XML_ELEMENT_NAME, null, null, fsd);
+		XMLUtils.writeDocumentToFile(subductionSectDataFile, doc0);
+
+		// calculate distances between each subsection
+		Map<IDPairing, Double> subSectionDistances = DeformationModelFetcher.calculateDistances(maxDistance, fsd);
+		Map<IDPairing, Double> reversed = Maps.newHashMap();
+        
+		//Dump distances		
+		for(IDPairing pair : subSectionDistances.keySet()) {
+			System.out.println("pair: "+pair+ ";  dist: "+subSectionDistances.get(pair));
+		}
+		assertEquals(1,2);
```

Comment:
Let's not commit failing tests to master.

Reply:
Maybe you could remove this file from the commit and leave it in your branch if you're still working on it?

Reply:
Right. In fact, I have already remove it from the branch.  Will have to push the changes.